buffer.go

```diff
@@ -9,6 +9,7 @@ type buffer struct {
 	metricCount int
 	flushSize   int
 	dryRun      bool
+	autoFlush   bool
```

Comment:
You're missing checking against this configuration on the `func (s *buffer) Put` and `func (s *buffer) PutAggregated` methods.

It needs to disable auto flushing by size as well (right now you're only disabling by time interval).